### PR TITLE
Create custom user package to obtain host user data

### PR DESCRIPTION
**Pull request by uri-weisman**

### CODE REVIEW

**Comment by uri-weisman** [comment]
fixes #118

File: resources/utils/user.go [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+type lineFunc func(line []byte) (v interface{}, err error)
+
+func GetUserNameFromID(uid uint32, userFilePath string) string {
+	u := strconv.FormatUint(uint64(uid), 10)
+	usr, err := lookupUserId(u, userFilePath)
+	if err != nil || usr == nil {
+		return ""
+	}
+
+	if usr.Name != "" {
```
Comment (uri-weisman) [inline resources/utils/user.go:75]: In case the user was not created with a friendly comment, `useradd -c "etcd"`, we'll return the username.

Comment (uri-weisman) [reply resources/utils/user.go:75]: e.g `list:x:38:38:Mailing List Manager:/var/list:/usr/sbin/nologin`
if "Mailing List Manager" doesn't exists use `list`

File: resources/utils/mocked_group_file.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+wheel:*:0:root
```
Comment (eyalkraft) [inline resources/utils/mocked_group_file.txt:1]: nice, let's move both mocks into `resources/utils/mock/`

**Review by eyalkraft — CHANGES_REQUESTED** [review]
Nice tests! small changes required

File: resources/fetchers/file_system_fetcher.go [hunk 3]
```diff
@@ -106,18 +108,15 @@ func FromFileInfo(info os.FileInfo, path string) (FileSystemResource, error) {
...
+
 	mod := strconv.FormatUint(uint64(info.Mode().Perm()), 8)
 	inode := strconv.FormatUint(stat.Ino, 10)
 
 	data := FileSystemResource{
 		FileName: info.Name(),
 		FileMode: mod,
-		Uid:      usr.Name,
-		Gid:      group.Name,
+		Uid:      utils.GetUserNameFromID(uid, UserFile),
```
[the remark this answers is not in the document]
Comment (uri-weisman) [reply resources/fetchers/file_system_fetcher.go]: https://github.com/elastic/security-team/issues/3889

**Review by eyalkraft — APPROVED** [review]
Test and ship

**Pull request merged**